duckduckgo/iOS
# Keep Kingfisher tab cache in sync with open tabs

Task/Issue URL: https://app.asana.com/0/414709148257752/1204375718000558/f
Tech Design URL:
CC:

**Description**:
One off cache clean-up of the tabs favicon cache + functionality added so that when a tab is closed, its favicon is deleted from tabs cache (as long as no other tabs have the same host)



**Steps to test this PR**:
Note: I branched off of updated Image Cache screen work so the Image Cache screen can be used for validation while testing 🙂 
1. To test the one-off clean up, ideally test from a device with orphaned tab favicons and confirm that after launching the app there are no longer any orphaned tab reported in the Debug > Image Cache screen (and confirm see no impact on performance in terms of opening + interacting with tabs)
2. Open a few tabs with different links for the same site
3. As you close each tab, confirm the favicon for the site remains in the Image Cache until the final tab for that site is closed. 



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
